bin/demo_sandbox.py

```diff
@@ -54,7 +58,19 @@
 nsims_n1 = args.nsims_n1 if not(args.nsims_n1 is None) else nsims_rdn0
 nsims_mf = args.nsims_mf if not(args.nsims_mf is None) else nsims_rdn0
 
-mg = solenspipe.LensingSandbox(fwhm_arcmin,noise_uk,dec_min,dec_max,res,lmin,lmax,mlmax,ests,add_noise=add_noise,verbose=True)
+# Placeholder until better way of inferring downgrade resolution
+if args.mask is not None:
+    mask = enmap.read_map(args.mask)
+    mask = enmap.downgrade(mask, int(res / (21600 / mask.shape[1])),
```

Comment:
See the latest commit -- the shape + wcs of the entire run is already based on the provided mask (LensingSandbox object is untouched in that regard), but the downgrading operation is now only called with an optional argument (where you can specify a common resolution of both the mask and map).

For the ACT sims, we tend to downgrade the input CMB maps before passing them into reconstruction, so the mask is naturally at the higher resolution -- wouldn't it be a better idea to set the shape / wcs of the run based on the input maps instead?

Reply:
It might be better to keep that outside of the sandbox class no? You could do something like

```
mask = enmap.downgrade(enmap.read_map(mask_file))
lensbox = LensingSandbox(mask=mask)
```

Reply:
This goes for apodization too, you could just do it before you initialize the sandbox. I think this is cleaner because you only want to add functionality to a class that is actually specific to the class object's state. If you can do an operation without having to know about the object's state, it's better to have that operation be outside the class.

Reply:
The latest commit keeps the downgrading and apodizing steps outside of the sandbox and also as optional parameters. Default behavior for the sandbox class is to infer the shape of the output maps as the shape of the mask, if one is passed in. Let me know if you have any other ideas or suggestions for improving the sandbox!